compiler/rustc_middle/src/ty/structural_impls.rs

```diff
@@ -1182,7 +1182,11 @@ impl<'tcx> TypeFoldable<'tcx> for ty::Const<'tcx> {
 
 impl<'tcx> TypeVisitable<'tcx> for ty::Const<'tcx> {
     fn visit_with<V: TypeVisitor<'tcx>>(&self, visitor: &mut V) -> ControlFlow<V::BreakTy> {
-        visitor.visit_const(*self)
+        visitor.visit_const(*self)?;
+        if let ty::ConstKind::Abstract(e) = self.kind() {
```

Comment:
that seems wrong, did you need to add that to fix any failing tests?

Reply:
Yea, adding this fixes 3 or so tests.

Reply:
hm, that's an issue with a visitor which doesn't recurse correctly, not with the `TypeVisitable` impl.

Right now abstract consts are visited twice.